penaltymodel_maxgap/tests/test_generation.py

```diff
@@ -1,124 +1,69 @@
+        for u, v in bqm.quadratic:
+            self.assertIn((u, v), graph.edges)
 
-    def check_quadratic_energy_ranges(self, quadratic, quadratic_energy_ranges):
-        for edge, bias in quadratic.items():
-            min_, max_ = quadratic_energy_ranges[edge]
-            self.assertGreaterEqual(bias, min_)
-            self.assertLessEqual(bias, max_)
+        # now solve for the thing
+        sampleset = dimod.ExactSolver().sample(bqm)
 
-    def check_generated_ising_model(self, feasible_configurations, decision_variables,
-                                    linear, quadratic, ground_energy, infeasible_gap):
-        """Check that the given Ising model has the correct energy levels"""
-        if not feasible_configurations:
-            return
+        # check that ground has 0 energy
+        if len(sampleset):
+            self.assertAlmostEqual(sampleset.first.energy, min(configurations.values()))
 
-        response = dimod.ExactSolver().sample_ising(linear, quadratic)
-
-        # samples are returned in order of energy
-        sample, ground = next(iter(response.data(['sample', 'energy'])))
-        gap = float('inf')
-
-        self.assertIn(tuple(sample[v] for v in decision_variables), feasible_configurations)
-
-        seen_configs = set()
-
-        for sample, energy in response.data(['sample', 'energy']):
+        # check gap and other energies
+        best_gap = float('inf')
+        seen = set()
+        for sample, energy in sampleset.data(['sample', 'energy']):
             config = tuple(sample[v] for v in decision_variables)
 
             # we want the minimum energy for each config of the decisison variables,
             # so once we've seen it once we can skip
-            if config in seen_configs:
+            if config in seen:
                 continue
 
-            if config in feasible_configurations:
-                self.assertAlmostEqual(energy, ground)
-                seen_configs.add(config)
+            if config in configurations:
+                self.assertAlmostEqual(energy, configurations[config])
+                seen.add(config)
             else:
-                gap = min(gap, energy - ground)
-
-        self.assertAlmostEqual(ground_energy, ground)
-        self.assertAlmostEqual(gap, infeasible_gap)
-
-    def test_trivial(self):
-        # this should test things like empty graphs and empty configs
-        pass
-
-    def test_basic(self):
-        """A typical use case, an AND gate on a chimera tile."""
-        graph = dnx.chimera_graph(1, 1, 4)
-        configurations = {(-1, -1, -1): 0,
-                          (-1, +1, -1): 0,
-                          (+1, -1, -1): 0,
-                          (+1, +1, +1): 0}
-        decision_variables = (0, 1, 2)
-        linear_energy_ranges = {v: (-2., 2.) for v in graph}
-        quadratic_energy_ranges = {(u, v): (-1., 1.) for u, v in graph.edges}
+                best_gap = min(best_gap, energy)
```

Comment:
Once you've found the `best_gap` you don't seem to do anything with it. Perhaps it should be tested against something?